charts/base/templates/deployment.yaml

```diff
@@ -94,8 +95,12 @@ spec:
       volumes:
         {{- range $index, $element := .Values.deployment.volumes }}
         - name: {{ coalesce $element.name (add $index 1) }}
+          {{- if $element.persistentVolumeClaim }}
           persistentVolumeClaim:
             claimName: {{ $element.persistentVolumeClaim.claimName }}
+          {{- end }}
+          secret:
```

Comment:
please put this config under condition like you did for persistentVolumeClaim case